client/components/app.jsx

```diff
@@ -1,14 +1,38 @@
 import React from 'react';
 import Header from './header';
 import ProductList from './product-list';
+import ProductDetails from './product-details';
 
 export default class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       message: null,
-      isLoading: true
+      isLoading: true,
+      view: {
+        name: 'catalog',
+        params: {}
+      }
     };
+
+    this.setView = this.setView.bind(this);
+  }
+
+  setView(params) {
+    let view;
+    if (this.state.view.name === 'catalog') {
+      view = {
+        name: 'details',
+        params: { productId: params.productId }
+      };
+      this.setState({ view });
+    } else {
+      view = {
+        name: 'catalog',
+        params: null
+      };
+      this.setState({ view });
+    }
```

Comment:
I recommend revising this method. You will not always want to go from the catalog to details. Instead, the method should accept which view to show and the params as arguments

```js
setView(name, params) {
  var view = {
    name: name,
    params: params
  }
  this.setState({ view });
```

When the method is called, it should be passed two arguments, the name of the view (a string), and the params (an object)